### DISCUSSION TITLE
Android Studio dependency resolval problems for mapsforge/vtm-android

### DISCUSSION THREAD

**Discussion by ImFelixprogram** [mapsforge/vtm#937] [Q&A] [answered]
Hello,

I have a problem with mapsforge/vtm since I updated to the newest Android Studio (2021.2.1 Patch 1),  and updated my project to the newest Gradle (7.3.3) and Android Gradle Plugin (7.2.1). When having the architecture specific imports like "org.mapsforge:vtm-android:0.18.0:natives-armeabi-v7a" in my build.gradle 
![Architecture specific imports](https://user-images.githubusercontent.com/50058400/175770303-67dab4fc-9dd6-44cd-bb55-35ab7aced381.PNG)
, Gradle builds it fine but Android Studio cannot find the main "org.mapsforge:vtm-android:0.18.0" dependency for imports
 ![Import errors](https://user-images.githubusercontent.com/50058400/175770301-7e7b13a6-82b8-4f0f-b4bc-8a3e2f5d9bcd.PNG).
 This is I think because Android Studio cannot process something after the version number like for example the "natives-armeabi-v7a" because it shows up as multiple of "org.mapsforge:vtm-android:0.18.0" instead in the project structure
 ![Screenshot of the project structure](https://user-images.githubusercontent.com/50058400/175770300-707c65d3-b722-40f9-95c2-71a1b3aeb1c1.PNG). Has anyone had that problem and figured out a solution to this? If yes, could you please tell me how that's fixed?

Greetings, Felixprogram

**Comment by devemux86** [comment, marked as answer]
You can include the native libs as .so files inside the app,
instead of using the jars.

**Reply by ImFelixprogram** [reply]
I'll try that, thank you. Where in the folder structure would I put the files? @devemux86

**Comment by devemux86** [comment]
There is information on the internet, in the Stack Overflow and Android documentation.
https://developer.android.com/studio/projects/gradle-external-native-builds#jniLibs

**Reply by ImFelixprogram** [reply]
Thank you very much, it worked.
